src/hotspot/share/opto/loopTransform.cpp

```diff
@@ -1201,11 +1201,12 @@ bool IdealLoopTree::policy_range_check(PhaseIdealLoop* phase, bool provisional,
 
       // Comparing trip+off vs limit
       Node *bol = iff->in(1);
-      if (bol->req() != 2) {
+      if (bol->req() < 2) {
         continue; // dead constant test
       }
       if (!bol->is_Bool()) {
-        assert(bol->Opcode() == Op_Conv2B, "predicate check only");
+        assert(bol->is_Opaque4() || bol->is_OpaqueInitializedAssertionPredicate(),
+               "Opaque node of non-null-check or of Initialized Assertion Predicate");
```

Comment:
We need this for Initialized `OpaqueInitializedAssertionPredicate` and `Opaque4` nodes for non-null-checks. This is covered by `testPolicyRangeCheck()`.

I did not try to come up with a test for Template Assertion Predicates. I think they cannot be found here since we eagerly remove them once they become useless. However, I'm planning to replace the `Opaque4` nodes for them anyways.